tests/resources/functions/ruby-3.1/tests.rb

```diff
@@ -1,36 +1,87 @@
 require 'httparty'
 require 'json'
 
-#    'req' variable has:
-#        'headers' - object with request headers
-#        'payload' - object with request body data
-#        'variables' - object with function variables
-#    'res' variable has:
-#        'send(text, status)' - function to return text response. Status code defaults to 200
-#        'json(obj, status)' - function to return JSON response. Status code defaults to 200
-#    
-#    If an error is thrown, a response with code 500 will be returned.
+def main(context)
+    action = context.req.headers['x-action'] || ''
 
-def main(req, res)
-    payload = JSON.parse(req.payload === '' ? '{}' : req.payload)
+    case action
+    when 'plaintextResponse'
+        return context.res.send('Hello World 👋')
+    when 'jsonResponse'
+        return context.res.json({ 'json': true, 'message': 'Developers are awesome.' })
+    when 'redirectResponse'
+        return context.res.redirect('https://github.com/')
+    when 'emptyResponse'
+        return context.res.empty()
+    when 'noResponse'
+        context.res.send('This should be ignored, as it is not returned.')
+        return nil
+    when 'doubleResponse'
+        context.res.send('This should be ignored.')
+        return context.res.send('This should be returned.')
+    when 'headersResponse'
+        return context.res.send('OK', 200, {
+            'first-header': 'first-value',
+            'second-header': context.req.headers['x-open-runtimes-custom-in-header'] || 'missing',
+            'x-open-runtimes-custom-out-header': 'third-value'
```

Comment:
What's the difference between first-header, second-header and x-open-runtime-custom-out-header? That third one is quite different from the others

Reply:
First is simply checking if response headers works.
Second is checking if we can read request headers.
Third is checking if reserved keyword (x-open-runtimes) headers are missing from response - intended.

Reply:
Alright, gotcha